pkg/solana/chainreader/batch.go

```diff
@@ -43,7 +50,26 @@ func doMethodBatchCall(ctx context.Context, client MultipleAccountGetter, bindin
 		if err != nil {
 			return nil, fmt.Errorf("failed to get address for %s account read: %w", call.ReadName, err)
 		}
-		keys[idx] = key
+
+		lvBinding, ok := binding.(latestValueBinding)
+		if ok {
+			results[idx] = batchResultWithErr{
+				address:      key.String(),
+				contractName: call.ContractName,
+				readName:     call.ReadName,
+				returnVal:    call.ReturnVal,
+			}
+
+			// is an event read
+			results[idx].err = lvBinding.GetLatestValue(ctx, call.Params, results[idx].returnVal)
```

Comment:
Is this "is an event read" comment right? How do we know it's an event read as opposed to an account read?

Reply:
if that's determined by whether the `latestValueBinding` interface is satisfied, maybe it should be called `eventBinding` instead?

Reply:
I agree it's confusing, but the account reads don't use `GetLatestValue`. They do batch data reads instead. I could rename these to offer better clarity.